include/zephyr/drivers/seu/seu.h

```diff
@@ -0,0 +1,410 @@
+/*
+ * Copyright (c) 2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
```

Comment:
please add Doxygen tags ``@file`` and ``@defgroup`` for header to properly show up in Doxygen documentation.
Would also be nice to have a doc page (even if it's just showing the "API Reference" for now) listed under https://docs.zephyrproject.org/latest/hardware/peripherals/index.html

Reply:
added information in the file